pydocstringformatter/_configuration/formatter_options.py

```diff
@@ -25,5 +28,5 @@ def register_arguments_formatters(
             action=BooleanOptionalAction,
             dest=name,
             help=f"Activate or deactivate {name}: {formatter.__doc__}",
-            default=not formatter.optional,
+            default=False if formatter.optional else None,
```

Comment:
Why is this now `False`?

Reply:
There's something wrong here, default can only be falsey whatever the value given.

Reply:
- If `formatter.optional==True`, `default=False`, same as before.  
- If `formatter.optional==False`, then whether to apply the filter or not depends on the value of `--style` passed on the command line and is not known when setting defaults.  [As mentioned above](https://github.com/DanielNoord/pydocstringformatter/pull/132#issuecomment-1200207883), I can set the default to a sentinel value (here `None` for simplicity) and use that to include or exclude formatters for a style *only* if they were not explicitly given on the command-line.

Reply:
This part is not specific to the numpy style formatter, so I think this should be handled elsewhere. We probably lack a test when the formatter is not optional and default should be true.

Reply:
> We probably lack a test when the formatter is not optional and default should be true.

Yeah, I was realizing that yesterday.  Should that be similar to the tests for `self.namespace.style`?

Reply:
I don't think that is necessary anymore.

In the current version optional checks also depend on the version.

So ``--numpydoc-name-type-spacing`` only works if you also provide ``--style=numpydoc``. That might be a bit counter intuitive, but I think of the `style` config as a way to tell `pydocstringformatter` which styles it should look out for and then apply any optional or non-optional formatters on them.